### ISSUE TITLE
I'm getting an error when calling query on search even though ES returns results (tire does not see them though)

### ISSUE THREAD

**Issue by chanced** [karmi/retire#228]
I have a single model indexed with Tire called "Users". Within that model, I have several nested elements (via has_many) that are also indexed. Indexing in Elastic Search works fine but Tire blows up when it parses the results.

I'm not sure exactly where the issue is occurring but I have tested the curl the search is generating and it works fine and has results.

Any/all help would be greatly appreciated. I've included the rails console output, the results from ES, as well as the User model and the search model.

https://gist.github.com/f31642d01ab9dd5a7f99

Thanks!

**Comment by vhyza** [comment]
Hello,

I tried to rewrite your gist to executable ruby script and it seems its working ok https://gist.github.com/1751827

Please can you post your mapping? `curl "http://localhost:9200/users/_mapping?pretty=true"` and some documents? `curl "http://localhost:9200/users/_search?q=*&size=5&pretty=1"`

**Comment by chanced** [comment]
@vhyza Thanks for taking the time to check out my issue. I've updated my gist to include both of what you were asking for.

**Comment by vhyza** [comment]
Your mapping seems ok. I think the problem is that this syntax:

```
search = Tire::Search::Search.new('users')
search.query { string('names:' + me.name) } unless name.empty?
search.filter :geo_distance, distance: distance_to_search, 'addresses.lat_lon' => lat_lon if search_distance?
return search
```

is available after commit 91961416f25b63d6641cadada71d469455ba4d90 which is not included in last `Tire` gem. You have to switch to block-style DSL. For example:

```
search = Tire.search 'users_tire_issue' do |search|
   search.query do |query|
     query.string "names: #{me.name}" unless me.name.empty?
   end

  search.filter :geo_distance, distance: "10mi", 'addresses.lat_lon' => me.lat_lon
end

search
```

until @karmi release new version of `Tire`

**Comment by chanced** [comment]
I moved it all to an overridden search method on the User object and structured it as you suggested. It worked like a charm, thanks man! This was a major PITA; thanks for helping me solve it man.
